src/coreclr/jit/importercalls.cpp

```diff
@@ -2759,6 +2759,32 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
+            case NI_System_Runtime_CompilerServices_RuntimeHelpers_IsReference:
+            {
+                assert(sig->sigInst.methInstCount == 1);
+                CORINFO_CLASS_HANDLE hClass = sig->sigInst.methInst[0];
+                retNode                     = gtNewIconNode(eeIsValueClass(hClass) ? 0 : 1);
```

Comment:
isn't eeIsValueClass `false` for unsafe pointers? (can be used via IL)

Reply:
it seems that CoreCLR and CG2/NAOT have different behavior here? cc @jkotas

Reply:
Generic instantiation arg cannot be a pointer, so there is no observable behavior difference. There are probably hundreds of places that make similar simplifying assumption, so I am not particularly worried about it.